base/src/main/java/org/aya/tyck/StmtTycker.java

```diff
@@ -228,7 +230,17 @@ public void tyckHeader(@NotNull Decl decl, @NotNull ExprTycker tycker) {
         assert structSig != null;
         var structLvl = structSig.result();
         var tele = tele(tycker, field.telescope, structLvl.isProp() ? null : structLvl);
+
+        if (!piPositivity(tele, structRef)) {
+          reporter.report(new PositivityError(field.sourcePos(), field, structRef.concrete));
+        }
```

Comment:
For this part I think it's unnecessary. The tele of the struct cannot refer to the struct itself. We already have error report about that IIRC